tendermint/src/serializers.rs

```diff
@@ -50,5 +50,5 @@ pub(crate) use raw_commit_sig::RawCommitSig;
 mod tests;
 
 mod custom;
-pub(crate) use custom::parse_non_empty_block_id;
-pub(crate) use custom::parse_non_empty_hash;
+pub use custom::parse_non_empty_block_id;
+pub use custom::parse_non_empty_hash;
```

Comment:
Do we have to publish these beyond the crate? Probably, yes as some might be used in the rpc crate now?

Reply:
Indeed, the rpc crate depends on a lot of these serialisers. Not happy with it. Also found the excessive use of `pub(crate)` is unnecessary as the top-level lib determines visibility beyond crate boundaries. Eager to hear how we can address this issue.

Reply:
If they are *only* used in the rpc lib, they could actually live there. Not sure yet.

Reply:
They are also used in `tendermint::block`.

Reply:
Maybe removing the net dependency outweighs this now publisized types. 

I might have mentioned that somewhere else before but ideally, the serialzation code would simply be abstracted away in the sense that anyone using the tendermint crate can e.g. simply call block::deserialize (or deserialize_json and deserialize_binary) and the that could internally use a dedicated serialization type serialize::block (and impl Froms between both) which uses methods / helpers like the above. Anyone (like the rpc crate) could simply use the core types and wouldn't need to care about serialization specific stuff. Does that make sense? Probably worth another discussion and later another PR.

Reply:
For now: maybe the `serializers` module should include a comment that we don't guarantee any backwards compatibility on the methods of that module and consider them an internal detail (use at you own risk alike).

Reply:
We need to lay out an explicit semantic versioning scheme and declare which modules we're covering and which we're not. Good to start tracking like this though thanks!